src/ui/components/Carousel/index.js

```diff
@@ -16,31 +18,80 @@ type PropTypes = {|
   sections: Array<React.Element<typeof CarouselSection>>,
 |};
 
-export const CarouselBase = ({ i18n, sections }: PropTypes) => {
-  if (!sections) {
-    throw new Error('sections are required for a Carousel component');
+type StateTypes = {|
+  hasRendered: boolean,
+|};
+
+export class CarouselBase extends React.Component {
+  constructor(props: PropTypes) {
+    super(props);
+
+    this.state = { hasRendered: false };
+  }
+
+  state: StateTypes;
+
+  // HACK: NukaCarousel uses inline styles to create the carousel effect, which
+  // work fine on the client but cause CSP issues during initial server render.
+  // To get around this, we re-render the component on the client side.
+  // We use componentDidMount to cause it to update on the client because this
+  // lifecycle method is only called on the client.
+  // See: https://github.com/mozilla/addons-frontend/issues/3349
+  componentDidMount() {
+    if (!this.state.hasRendered) {
+      log.debug('Re-rendering Carousel to avoid CSP issues.');
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({ hasRendered: true });
+    }
   }
 
-  return (
-    <Card className="Carousel">
-      <NukaCarousel
-        autoplay
-        autoplayInterval={4000}
-        cellAlign={getDirection(i18n.lang) === 'ltr' ? 'left' : 'right'}
-        cellSpacing={10}
-        framePadding="0 10%"
-        frameOverflow="visible"
-        slidesToShow={1}
-        slidesToScroll={1}
-        slideWidth={1}
+  props: PropTypes;
+
+  render() {
+    const { i18n, sections } = this.props;
+
+    if (!sections) {
+      throw new Error('sections are required for a Carousel component');
+    }
+
+    return (
+      <Card
+        className={classNames('Carousel', {
+          'Carousel--first-render-wrapper': !this.state.hasRendered,
+        })}
       >
-        {(sections.map((section) => {
-          return section;
-        }))}
-      </NukaCarousel>
-    </Card>
-  );
-};
+        {this.state.hasRendered ? (
+          <NukaCarousel
+            autoplay
+            autoplayInterval={4000}
+            cellAlign={getDirection(i18n.lang) === 'ltr' ? 'left' : 'right'}
```

Comment:
Didn't see this before, but shouldn't we have constants for that? It would avoid case issues. If so, then I could open a PR because it is unrelated.

https://github.com/mozilla/addons-frontend/blob/92b83273ba0c86ab750e92faf1c8e5a50c033aa8/src/core/i18n/utils.js#L104-L106

Reply:
See 3362